### ISSUE TITLE
Migrate state-related classes from `swirlds-platform-core` to `swirlds-state-impl`

### ISSUE THREAD

**Issue by imalygin** [hashgraph/hedera-services#16223]
Classes to be moved:
- `MerkleStateRoot`
- `MerkleRoot` - this class should be removed as a part of #15638 . But if this task is not completed before the current one, it should be moved. 
- `StateChangeListener`
- `MerkleTreeSnapshotReader` / `MerkleTreeSnapshotWriter` and their dependencies
- `MerkleStateUtils`
- `MerkleRootSnapshotMetrics`

Note that all the dependencies of the abovementioned classes are supposed to belong to `com.swirlds.platform.state`.

**Comment by imalygin** [comment]
This PR - https://github.com/hashgraph/hedera-services/issues/16322 - partially addresses this ticket but it has to be merged first.

**Comment by thenswan** [comment]
- `MerkleStateRoot` is already in `swirlds-state-impl`;
- `MerkleRoot` requires dependencies from `swirlds-platform-core`, moving it into `swirlds-state-impl` will introduce circular dependency **(maybe we can just remove it)**;
- `StateChangeListener` is already in `swirlds-state-api`;
- `MerkleTreeSnapshotReader` and `MerkleTreeSnapshotWriter` are already in `swirlds-state-impl`;
- `MerkleStateUtils` depends on `ConsensusSnapshot` which is in `swirlds-platform-core` **(we can remove dependency and move the class)**;
- `MerkleRootSnapshotMetrics` is already in `swirlds-state-impl`;

**Comment by imalygin** [comment]
Upon further investigation we figured out that it doesn't make sense to move `MerkleStateUtils` as it's relevant for Platform code only. As there is a separate ticket for `MerkleRoot` deletion, we can close this one, as there is nothing left to do here.
